controllers/storageclient_controller.go

```diff
@@ -203,9 +205,57 @@ func (r *StorageClientReconciler) reconcilePhases() (ctrl.Result, error) {
 		}
 	}
 
+	if err := r.reconcileClusterResourceQuota(externalClusterClient); err != nil {
+		return reconcile.Result{}, err
+	}
+
 	return reconcile.Result{}, nil
 }
 
+func (r *StorageClientReconciler) reconcileClusterResourceQuota(externalClusterClient *providerClient.OCSProviderClient) error {
+
+	scResponse, err := externalClusterClient.GetStorageConfig(r.ctx, r.storageClient.Status.ConsumerID)
+	if err != nil {
+		return fmt.Errorf("failed to get StorageConfig for setting quota: %v", err)
+	}
+	for _, eResource := range scResponse.ExternalResource {
+		if eResource.Kind == "ClusterResourceQuota" {
+
+			data := map[string]string{}
+			err := json.Unmarshal(eResource.Data, &data)
+			if err != nil {
+				return fmt.Errorf("failed to unmarshal data: %v", err)
+			}
+			clusterResourceQuotaSpec, ok := data["spec"]
+			if !ok {
+				return fmt.Errorf("no spec key found in the map")
+			}
+			fmt.Print(clusterResourceQuotaSpec)
+			var clusterResourceQuota quotav1.ClusterResourceQuota
+
+			err = json.Unmarshal([]byte(clusterResourceQuotaSpec), &clusterResourceQuota)
+			if err != nil {
+				return fmt.Errorf("error: %v", err)
+			}
+			fmt.Printf("%+v\n", clusterResourceQuota)
+
+			_, err = controllerutil.CreateOrUpdate(r.ctx, r.Client, &clusterResourceQuota, func() error {
+				// cluster scoped resource owning namespace scoped resource which allows garbage collection
+				if err := r.own(&clusterResourceQuota); err != nil {
```

Comment:
miss leading comment, isn't clusterresourcequota a cluster scoped resource?

Reply:
I copied this from some other controller and missed removing this comment. Thank you for noticing